[lldb] Remove limit on max memory read size

jasonmolenda opened the pull request:
`memory read` will return an error if you try to read more than 1k bytes in a single command, instructing you to set
`target.max-memory-read-size` or use `--force` if you intended to read more than that.  This is a safeguard for a command where people are being explicit about how much memory they would like lldb to read (either to display, or save to a file) and is an annoyance every time you need to read more than a small amount.  If someone confuses the --count argument with the start address, lldb may begin dumping gigabytes of data but I'd rather that behavior than requiring everyone to special-case their way around a common use case.

I don't want to remove the setting because many people have added (much larger) default max read sizes to their ~/.lldbinit files after hitting this behavior.  Another option would be to stop reading/using the value in Target.cpp, but I see no harm in leaving the setting if someone really does prefer to have a small cap on their memory read size.

jasonmolenda committed:
[lldb] Remove limit on max memory read size

`memory read` will return an error if you try to read more than 1k
bytes in a single command, instructing you to set
`target.max-memory-read-size` or use `--force` if you intended to
read more than that.  This is a safeguard for a command where people
are being explicit about how much memory they would like lldb to
read (either to display, or save to a file) and is an annoyance
every time you need to read more than a small amount.  If someone
confuses the --count argument with the start address, lldb may begin
dumping gigabytes of data but I'd rather that behavior than requiring
everyone to special-case their way around a common use case.

jasonmolenda commented:
(the setting is a UInt64, but Target.cpp reads it and returns a uint32_t, so I only did UINT32_MAX value)

jasonmolenda commented:
lol my hubris is on display, the PR testing hit a test failure in ... TestMemoryReadMaximumSize.py which I wrote & merged earlier today, and assumes there is a limit of 1024. ;)

medismailben commented on lldb/source/Target/TargetProperties.td:
```diff
@@ -102,7 +102,7 @@ let Definition = "target" in {
     Desc<"Maximum number of characters to show when using %s in summary strings.">;
   def MaxMemReadSize: Property<"max-memory-read-size", "UInt64">,
-    DefaultUnsignedValue<1024>,
+    DefaultUnsignedValue<4294967295>,
```
suggestion: Either make it hex, or instead of providing a default value here, initialize the CommandOption to UINT32_MAX

jasonmolenda committed:
Change to using a hex value in .td, fix API test

jasonmolenda commented:
Updated patch to a hex constant in the .td file as per Ismail's suggestion, fixed the new TestMemoryReadMaximumSize.py test to work with the new default.

medismailben reviewed (approved):
LGTM

jasonmolenda commented:
both of the linux PR tests failed in Unwind/trap_frame_sym_ctx.test, I could have sworn the only test failure I had before was in TestMemoryReadMaximumSize.py, I don't see how this PR breaks that test case. Hmm.

medismailben commented:
> both of the linux PR tests failed in Unwind/trap_frame_sym_ctx.test, I could have sworn the only test failure I had before was in TestMemoryReadMaximumSize.py, I don't see how this PR breaks that test case. Hmm.

I believe that failure is unrelated to your PR since another user have the exact same issue in theirs: https://github.com/llvm/llvm-project/pull/105757. The issue seems to be resolved by https://github.com/llvm/llvm-project/pull/105695.

jasonmolenda commented:
> > both of the linux PR tests failed in Unwind/trap_frame_sym_ctx.test, I could have sworn the only test failure I had before was in TestMemoryReadMaximumSize.py, I don't see how this PR breaks that test case. Hmm.
> 
> I believe that failure is unrelated to your PR since another user have the exact same issue in theirs: #105757. The issue seems to be resolved by #105695.

thanks!  I want to leave this PR open until next week in case @clayborg has an opinion (this is def his orig code), but I won't look into that fail any further.

jasonmolenda committed:
Merge branch 'main' into remove-memory-read-default-1024

jasonmolenda commented:
@clayborg what do you think about removing the memory limit from memory read?  I know this is code you originally wrote, and I might not be thinking of a way for someone to do a huge memory read unintentionally.  But it seems like this cap just gets in people's way more than anything else.

(I have complaints about our other safety limits like the 1024 byte c-string formatter cap which is really easy to hit with long strings, but those are more debatable ;)
